### ISSUE TITLE
[PR #520] cleartext sensitive info in sqlitetest.rs

### ISSUE THREAD

**Issue by il10241024** [constructorfabric/gears-rust#3911]
> 🔗 **Mirrored PR** [cyberfabric/cyberware-rust#520](https://github.com/cyberfabric/cyberware-rust/pull/520) | **Author:** dominic1988-lgtm | **Opened:** 2026-02-09T07:55:06Z | **Status:** merged on 2026-02-13T10:28:20Z
> *GitHub API does not allow setting PR author or timestamps — attribution preserved here.*

---

## Summary by CodeRabbit

* **Tests**
  * Improved SQLite pragma tests (WAL and busy_timeout) with clearer failure diagnostics: when an unexpected connection error occurs the failure message now reports the concrete error type (rather than raw debug output). No behavioral changes—only clearer, more actionable test messages and minor formatting refinements.

---

**Comment by il10241024** [comment]
**qodo-code-review[bot]** commented on 2026-02-09T07:55:18Z:

---

<h3>Review Summary by Qodo</h3>

Remove sensitive error details from SQLite test panic messages

<code>🐞 Bug fix</code>

<img src="https://www.qodo.ai/wp-content/uploads/2025/11/light-grey-line.svg" height="10%" alt="Grey Divider">

<h3>Walkthroughs</h3>

<details open>
<summary>Description</summary>

<br/>

<pre>
• Remove sensitive error details from panic messages in tests
• Replace full error debug output with error type names only
• Prevent exposure of potentially sensitive information in test failures
• Improve error diagnostics while maintaining privacy
</pre>
</details>

<details>
<summary>Diagram</summary>

<br/>

>
```mermaid
flowchart LR
  A["Panic messages with full error details"] -- "Replace with error type names" --> B["Sanitized panic messages"]
  C["Debug output {err:?}"] -- "Use type_name_of_val" --> D["Error type identifier"]
  E["Full result debug {result:?}"] -- "Match and extract type" --> F["Specific error variant name"]
```



</details>




<img src="https://www.qodo.ai/wp-content/uploads/2025/11/light-grey-line.svg" height="10%" alt="Grey Divider">

<h3>File Changes</h3>

<details>
<summary>1. libs/modkit-db/tests/sqlite/sqlite_tests.rs
 <code>🐞 Bug fix</code> <code> +35/-5 </code>  </summary>  

<br/>

>Sanitize error details in test panic messages
><pre>
>• Replaced full error debug output <b><i>{err:?}</i></b> with <b><i>std::any::type_name_of_val(&amp;err)</i></b> in WAL pragma
>  test
>• Replaced full result debug output <b><i>{result:?}</i></b> with pattern-matched error type names in three test
>  functions
>• Added match expressions to extract specific <b><i>DbError</i></b> variant names instead of printing full error
>  details
>• Applied consistent error sanitization across <b><i>test_wal_pragma_validation</i></b>,
>  <b><i>test_busy_timeout_pragma_validation</i></b> test cases
></pre>
>
><a href='https://github.com/constructorfabric/cyberfabric-core/issues/520/files#diff-296123d5a83dabd26a64de6a01f2ef8cd11d20dbe613b4e83e670402f1dd5b97'> libs/modkit-db/tests/sqlite/sqlite_tests.rs
 </a>
<hr/>

</details>

<img src="https://www.qodo.ai/wp-content/uploads/2025/11/light-grey-line.svg" height="10%" alt="Grey Divider">


<a href="https://www.qodo.ai"><img src="https://www.qodo.ai/wp-content/uploads/2025/03/qodo-logo.svg" width="80" alt="Qodo Logo"></a>

**Comment by il10241024** [comment]
**coderabbitai[bot]** commented on 2026-02-09T07:55:28Z:

---




<details>
<summary>📝 Walkthrough</summary>

## Walkthrough

Tests for SQLite pragma validation (WAL and `busy_timeout`) now report the concrete runtime error type using `type_name_of_val(&err)` in panic messages when a connection unexpectedly errors; no control flow or other logic was changed.

## Changes

|Cohort / File(s)|Summary|
|---|---|
|**SQLite Test Error Diagnostics** <br> `libs/modkit-db/tests/sqlite/sqlite_tests.rs`|Adjusted panic messages in WAL and `busy_timeout` pragma tests to display the concrete error type via `type_name_of_val(&err)` instead of the error's `Debug` output. No behavior or control flow changes.|

## Estimated code review effort

🎯 1 (Trivial) | ⏱️ ~2 minutes

## Poem

> 🐇 I hopped through tests with eager eyes,  
> Swapped the debug for a type surprise,  
> Now when things panic,  
> The error's name will yank—  
> A crisp little hint beneath the skies. 🥕

</details>






<details>
<summary>🚥 Pre-merge checks | ✅ 3</summary>

<details>
<summary>✅ Passed checks (3 passed)</summary>

|     Check name     | Status   | Explanation                                                                                                                                                             |
| :----------------: | :------- | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
|  Description Check | ✅ Passed | Check skipped - CodeRabbit’s high-level summary is enabled.                                                                                                             |
|     Title check    | ✅ Passed | The title accurately describes the main change: removing cleartext sensitive information from error messages in sqlite tests by replacing debug format with type names. |
| Docstring Coverage | ✅ Passed | Docstring coverage is 100.00% which is sufficient. The required threshold is 80.00%.                                                                                    |

</details>

<sub>✏️ Tip: You can configure your own custom pre-merge checks in the settings.</sub>

</details>





<details>
<summary>✨ Finishing touches</summary>

- [ ]  📝 Generate docstrings
<details>
<summary>🧪 Generate unit tests (beta)</summary>

- [ ]    Create PR with unit tests
- [ ]    Post copyable unit tests in a comment

</details>

</details>




---

No actionable comments were generated in the recent review. 🎉



---



<sub>Comment `@coderabbitai help` to get the list of available commands and usage tips.</sub>

**Comment by il10241024** [comment]
**qodo-code-review[bot]** commented on 2026-02-09T08:00:11Z:

---

<h3>Code Review by Qodo</h3>

<code>🐞 Bugs (1)</code>  <code>📘 Rule violations (0)</code>  <code>📎 Requirement gaps (0)</code>

<img src="https://www.qodo.ai/wp-content/uploads/2025/11/light-grey-line.svg" height="10%" alt="Grey Divider">

<br/>

<img src="https://www.qodo.ai/wp-content/uploads/2026/01/review-recommended.png" height="20" alt="Remediation recommended">

<details>
<summary>  1.  Error-type panic regression <code>🐞 Bug</code> <code>✧ Quality</code></summary>

<br/>

> <details open>
><summary>Description</summary>
><br/>
>
><pre>
>• In the InvalidSqlitePragma negative tests, the new else-branch panics only print a coarse
>   <b><i>error_type</i></b> label and discard the underlying error payload.
>• This regresses from the prior <b><i>{result:?}</i></b> output and hides details that are already present in
>   <b><i>DbError::InvalidSqlitePragma.message</i></b> (e.g., the invalid value).
>• The added match boilerplate also includes an <b><i>Ok(_)</i></b> arm even though <b><i>assert!(result.is_err())</i></b>
>   makes success unreachable.
></pre>
></details>

> <details open>
><summary>Code</summary>
><br/>
>
><code>[libs/modkit-db/tests/sqlite/sqlite_tests.rs[R415-425]](https://github.com/constructorfabric/cyberfabric-core/issues/520/files#diff-296123d5a83dabd26a64de6a01f2ef8cd11d20dbe613b4e83e670402f1dd5b97R415-R425)</code>
>
>```diff
>+         let error_type = match result {
>+            Ok(_) => "Ok (unexpected success)",
>+            Err(DbError::InvalidConfig(_)) => "InvalidConfig",
>+            Err(DbError::ConfigConflict(_)) => "ConfigConflict",
>+            Err(DbError::UnknownSqlitePragma(_)) => "UnknownSqlitePragma",
>+            Err(DbError::Sqlx(_)) => "SqlxError",
>+            Err(DbError::Sea(_)) => "SeaOrmError",
>+            Err(DbError::Io(_)) => "IoError",
>+            Err(_) => "OtherError",
>+        };
>+        panic!("Expected InvalidSqlitePragma error, got: {error_type}");
>```
></details>

> <details >
><summary>Evidence</summary>
><br/>
>
><pre>
>The else branches now intentionally reduce output to a string label, even though the library
>constructs informative messages for invalid pragmas. Those details are useful for debugging and are
>lost with the new approach.
></pre>
>
> <code>[libs/modkit-db/tests/sqlite/sqlite_tests.rs[410-426]](https://github.com/constructorfabric/cyberfabric-core/blob/86fc18d670eaaacecd1a7477115e320120819b01/libs/modkit-db/tests/sqlite/sqlite_tests.rs/#L410-L426)</code>
> <code>[libs/modkit-db/tests/sqlite/sqlite_tests.rs[490-506]](https://github.com/constructorfabric/cyberfabric-core/blob/86fc18d670eaaacecd1a7477115e320120819b01/libs/modkit-db/tests/sqlite/sqlite_tests.rs/#L490-L506)</code>
> <code>[libs/modkit-db/src/options.rs[192-231]](https://github.com/constructorfabric/cyberfabric-core/blob/86fc18d670eaaacecd1a7477115e320120819b01/libs/modkit-db/src/options.rs/#L192-L231)</code>
></details>

> <details>
><summary>Agent prompt</summary>
><br/>
>
>````
>The issue below was found during a code review. Follow the provided context and guidance below and implement a solution
>
>### Issue description
>The negative-path tests for invalid SQLite pragmas now panic with only a coarse `error_type` string. This discards the actual error payload (variant data + message), regressing debuggability and adding boilerplate.
>
>### Issue Context
>The library constructs informative error messages (e.g., includes the invalid value) in `DbError::InvalidSqlitePragma`. Tests should preserve that output when expectations fail.
>
>### Fix Focus Areas
>- libs/modkit-db/tests/sqlite/sqlite_tests.rs[410-426]
>- libs/modkit-db/tests/sqlite/sqlite_tests.rs[490-506]
>- libs/modkit-db/tests/sqlite/sqlite_tests.rs[530-546]
>
>### Suggested change
>Refactor to something like:
>```rust
>let err = result.expect_err(&quot;expected error&quot;);
>match err {
>    DbError::InvalidSqlitePragma { key, message } =&gt; { /* assertions */ }
>    other =&gt; panic!(&quot;Expected InvalidSqlitePragma, got: {other:?}&quot;),
>}
>```
>This keeps output actionable without extra matching boilerplate or unreachable arms.
>````
> <code>ⓘ Copy this prompt and use it to remediate the issue with your preferred AI generation tools</code>
></details>

<hr/>
</details>


<img src="https://www.qodo.ai/wp-content/uploads/2025/11/light-grey-line.svg" height="10%" alt="Grey Divider">
<pre>ⓘ The new review experience is currently in Beta. <a href="https://docs.qodo.ai/qodo-documentation/code-review">Learn more</a></pre>
<img src="https://www.qodo.ai/wp-content/uploads/2025/11/light-grey-line.svg" height="10%" alt="Grey Divider">




<a href="https://www.qodo.ai"><img src="https://www.qodo.ai/wp-content/uploads/2025/03/qodo-logo.svg" width="80" alt="Qodo Logo"></a>

**Comment by il10241024** [comment]
**coderabbitai[bot]** submitted review **COMMENTED** on 2026-02-09T08:01:27Z:

---

**Actionable comments posted: 1**

> [!CAUTION]
> Some comments are outside the diff and can’t be posted inline due to platform limitations.
> 
> 
> 
> <details>
> <summary>⚠️ Outside diff range comments (1)</summary><blockquote>
> 
> <details>
> <summary>libs/modkit-db/tests/sqlite/sqlite_tests.rs (1)</summary><blockquote>
> 
> `410-425`: _⚠️ Potential issue_ | _🔴 Critical_
> 
> **Avoid using `result` after it has been moved.**
> 
> `if let Err(...) = result` moves `result`; the subsequent `match result` in the `else` branch will not compile. Borrow the result instead.
> 
> <details>
> <summary>✅ Borrow to avoid move</summary>
> 
> ```diff
> -    if let Err(DbError::InvalidSqlitePragma { key, message }) = result {
> +    if let Err(DbError::InvalidSqlitePragma { key, message }) = &result {
>          assert_eq!(key, "wal");
>          assert!(message.contains("true/false/1/0"));
>      } else {
> -         let error_type = match result {
> +        let error_type = match &result {
>              Ok(_) => "Ok (unexpected success)",
>              Err(DbError::InvalidConfig(_)) => "InvalidConfig",
>              Err(DbError::ConfigConflict(_)) => "ConfigConflict",
>              Err(DbError::UnknownSqlitePragma(_)) => "UnknownSqlitePragma",
>              Err(DbError::Sqlx(_)) => "SqlxError",
>              Err(DbError::Sea(_)) => "SeaOrmError",
>              Err(DbError::Io(_)) => "IoError",
>              Err(_) => "OtherError",
>          };
>          panic!("Expected InvalidSqlitePragma error, got: {error_type}");
>      }
> ```
> </details>
> 
> </blockquote></details>
> 
> </blockquote></details>

<details>
<summary>🤖 Fix all issues with AI agents</summary>

```
In `@libs/modkit-db/tests/sqlite/sqlite_tests.rs`:
- Line 383: Fix the malformed panic string in the test: the panic invocation
that references wal_value currently leaves an unmatched single quote in the
format string. Update the panic call so the wal_value is inserted via a format
placeholder with its closing quote (e.g. "'{}'") and pass wal_value as the first
argument and std::any::type_name_of_val(&err) as the second, ensuring the format
string and arguments match (the call around wal_value and
std::any::type_name_of_val(&err) in sqlite_tests.rs).
```

</details>

**Comment by il10241024** [comment]
**coderabbitai[bot]** reviewed `libs/modkit-db/tests/sqlite/sqlite_tests.rs` line 383 on 2026-02-09T08:01:26Z:

---

_⚠️ Potential issue_ | _🟡 Minor_

**Fix the missing closing quote in the WAL panic message.**

The message currently renders with an unmatched quote, which is noisy in test output.

<details>
<summary>🔧 Proposed fix</summary>

```diff
-                panic!("Expected successful connection with WAL value '{wal_value}, error type: {}", std::any::type_name_of_val(&err));
+                panic!("Expected successful connection with WAL value '{wal_value}', error type: {}", std::any::type_name_of_val(&err));
```
</details>



<details>
<summary>📝 Committable suggestion</summary>

> ‼️ **IMPORTANT**
> Carefully review the code before committing. Ensure that it accurately replaces the highlighted code, contains no missing lines, and has no issues with indentation. Thoroughly test & benchmark the code to ensure it meets the requirements.

```suggestion
                panic!("Expected successful connection with WAL value '{wal_value}', error type: {}", std::any::type_name_of_val(&err));
```

</details>



<details>
<summary>🤖 Prompt for AI Agents</summary>

```
In `@libs/modkit-db/tests/sqlite/sqlite_tests.rs` at line 383, Fix the malformed
panic string in the test: the panic invocation that references wal_value
currently leaves an unmatched single quote in the format string. Update the
panic call so the wal_value is inserted via a format placeholder with its
closing quote (e.g. "'{}'") and pass wal_value as the first argument and
std::any::type_name_of_val(&err) as the second, ensuring the format string and
arguments match (the call around wal_value and std::any::type_name_of_val(&err)
in sqlite_tests.rs).
```

</details>





✅ Addressed in commit 6b42e4a

**Comment by il10241024** [comment]
**Bechma** reviewed `libs/modkit-db/tests/sqlite/sqlite_tests.rs` line 415 on 2026-02-09T08:48:45Z:

---

While the idea is not bad, I don't like it. This change is unnecesary pedantic for a test.
I would say to keep both approaches you presented in this PR:
std::any::type_name_of_val(&err)
AND
{result:?}

in tests there should not be a problem to print error messages, as we are using mock data all the time

**Comment by il10241024** [comment]
**dominic1988-lgtm** reviewed `libs/modkit-db/tests/sqlite/sqlite_tests.rs` line 415 on 2026-02-09T08:55:24Z:

---

Updated
